lib/netext/httpext/transport.go

```diff
@@ -0,0 +1,147 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2019 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package httpext
+
+import (
+	"net"
+	"net/http"
+	"net/http/httptrace"
+	"strconv"
+
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/lib/netext"
+	"github.com/loadimpact/k6/stats"
+	"github.com/pkg/errors"
+)
+
+// transport is an implemenation of http.RoundTripper that will measure different metrics for each
+// roundtrip
+type transport struct {
```

Comment:
Still not sure why you un-exported this, especially after you bothered to properly comment all of the methods :smile:

Reply:
for me being exported is only acceptable when it will be used outside of the package .. it isn't any more and ... I doubt it will ever be again ;) if it becomes we can always export it again